### ISSUE TITLE
Regarding the issue with reproducing the training process

### ISSUE THREAD

**Issue by qsc1103** [Hon-Wong/VoRA#12]
Thank you for your excellent work!
I encountered some issues during reproduction. After completing the pretraining, merging LoRA, and finetuning steps, all the evaluation metrics on MME are zero. What could be the reason for this?

![Image](https://github.com/user-attachments/assets/98a2a099-391a-46a4-b050-deb165509061)

**Comment by Hon-Wong** [comment]
Hi, could you please provide more context? Specifically, it will be helpful to know your training loss, config, and any other details. It seems that your model may be collapsing and struggling to follow instructions for some unknown reason. For reference, a random guess on MME_p yields a score of around 500.

Thank you!

**Comment by qsc1103** [comment]
Thank you for your response. I'd like to first confirm whether it's necessary to merge LoRA again after finetuning. In the training process, it was only mentioned that LoRA should be merged before finetuning.

**Comment by Hon-Wong** [comment]
In finetuning, LoRA is not utilized because the LoRA parameters have already been merged after pretraining. Thus, for finetuning, the model only consists of an LLM together with a shallow vision embedding layer, where all parameters are unfrozen.

**Comment by qsc1103** [comment]
After merging LoRA, the LoRA parameters should have been integrated, as shown in Figure 1. However, as required, after finetuning, the LoRA modules reappear, as shown in Figure 2. There shouldn't be any LoRA layers at this point, right? What could be causing this issue?

![Image](https://github.com/user-attachments/assets/8745f9cf-fe60-43a9-befa-e15a8ebbcb16)

![Image](https://github.com/user-attachments/assets/aadcac33-ec12-497f-b950-8b17d27f9c0e)

**Comment by Hon-Wong** [comment]
Apologies for the confusion. We found a bug in the configs/finetuning.yaml: the parameter r should be set to -1. This has been corrected in our latest version. Thank you for your understanding.

**Comment by qsc1103** [comment]
After fixing the bug and retraining the finetuning stage, the evaluation metrics on MME are still abnormal (all zeros). What could be the cause of this issue?
Also, when training with the modified configs/finetuning.yaml, the resulting model weights are shown in Figure 1, while the weights from Hugging Face are shown in Figure 2. Is the difference in total_size due to storage precision?

![Image](https://github.com/user-attachments/assets/bb25d4ae-f7fa-4b8c-b761-eabfd87b2ceb)

![Image](https://github.com/user-attachments/assets/e561db5c-2668-409a-a129-a3f32f48b0ab)

**Comment by Hon-Wong** [comment]
I think that you can check the pre-training process first to see if the base model works as expected. To do so, just follow the code [here](https://huggingface.co/Hon-Wong/VoRA-7B-Base) to test a few images and review the output. 

Thank you!

**Comment by Hasnat79** [comment]
Hi @qsc1103 , 
I am also trying to replicate this work, but due to resources, I am trying to reproduce this work data-wise minimally. 

Could you please share how did you download and organized the annotation path and images  of llaVA in the server? 

I'm facing some issues, for example, I downloaded the llava_v1_5_mix665k.json from HF, but setting that json file in the config is giving me some formatting error from the __init__ of VoRADataset class while trying to load the json. 


I would really appreciate your help on this. 

Thanks.

**Comment by Hasnat79** [comment]
Hi @Hon-Wong, 

Thank you for giving us this new paradigm.
I am trying to replicate your work with limited data samples. Could you please give me example of how the json files of annotation looks like, and how would I build them with VoRA version of datasets, and how did you guys filtered specific set of images form big datasets (any data curation scripts if available) ? 

Thanks
Hasnat
